backend/src/main/java/ai/verta/modeldb/configuration/AppConfigBeans.java

```diff
@@ -0,0 +1,412 @@
+          appContext.registeredBean("nfsController", NFSController.class);
+          String rootDir = artifactStoreConfig.getNFS().getNfsRootPath();
+          LOGGER.trace("NFS server root path {}", rootDir);
+          final var props = new FileStorageProperties();
+          props.setUploadDir(rootDir);
+          artifactStoreService = new NFSService(props);
+          break;
+        default:
+          throw new ModelDBException("Configure valid artifact store name in config.yaml file.");
+      }
+      // ------------- Finish Initialize Cloud storage base on configuration ------------------
+
+      LOGGER.info(
+          "ArtifactStore service initialized and resolved storage dependency before server start");
+      return artifactStoreService;
+    } else {
+      LOGGER.info("Artifact store service is disabled.");
+      return null;
+    }
+  }
+
+  public static boolean migrate(MDBConfig mdbConfig, JdbiUtil jdbiUtil, Executor handleExecutor)
+      throws Exception {
+    var liquibaseMigration =
+        Boolean.parseBoolean(
+            Optional.ofNullable(System.getenv(CommonConstants.ENABLE_LIQUIBASE_MIGRATION_ENV_VAR))
+                .orElse("false"));
+
+    var modelDBHibernateUtil = ModelDBHibernateUtil.getInstance();
+    modelDBHibernateUtil.initializedConfigAndDatabase(mdbConfig, mdbConfig.getDatabase());
+
+    if (liquibaseMigration) {
+      LOGGER.info("Liquibase migration starting");
+      jdbiUtil.runLiquibaseMigration();
+      LOGGER.info("Liquibase migration done");
+
+      modelDBHibernateUtil.createOrGetSessionFactory(mdbConfig.getDatabase());
+      LOGGER.info("Old code migration using hibernate starting");
+      modelDBHibernateUtil.runMigration(mdbConfig.getDatabase(), mdbConfig.migrations);
+      LOGGER.info("Old code migration using hibernate done");
+
+      LOGGER.info("Code migration starting");
+      jdbiUtil.runMigration(handleExecutor);
+      LOGGER.info("Code migration done");
+
+      boolean runLiquibaseSeparate =
+          Boolean.parseBoolean(
+              Optional.ofNullable(System.getenv(CommonConstants.RUN_LIQUIBASE_SEPARATE))
+                  .orElse("false"));
+      LOGGER.trace("run Liquibase separate: {}", runLiquibaseSeparate);
+      if (runLiquibaseSeparate) {
+        return true;
+      }
+    }
+
+    modelDBHibernateUtil.createOrGetSessionFactory(mdbConfig.getDatabase());
+    jdbiUtil.setIsReady();
+
+    return false;
+  }
+
+  @Bean
+  public CommandLineRunner commandLineRunner(ApplicationContext ctx) {
+    return args -> {
+      try {
+        LOGGER.info("Backend server starting.");
+        final var logger =
+            java.util.logging.Logger.getLogger("io.grpc.netty.NettyServerTransport.connections");
+        logger.setLevel(Level.WARNING);
+
+        final var config = ctx.getBean(MDBConfig.class);
+
+        // Configure server
+        final var services = ctx.getBean(ServiceSet.class);
+        final var jdbiUtil = ctx.getBean(JdbiUtil.class);
```

Comment:
rather than getting these from the context, we should put them as parameters and let spring do the work

Reply:
Resolved